homeassistant/helpers/entity_registry.py

```diff
@@ -0,0 +1,123 @@
+"""Provide a registry to track entity IDs."""
+import asyncio
+from collections import namedtuple, OrderedDict
+from itertools import chain
+import logging
+import os
+
+from ..core import callback, split_entity_id
+from ..util import ensure_unique_string, slugify
+from ..util.yaml import load_yaml, save_yaml
+
+PATH_REGISTRY = 'entity_registry.yaml'
+SAVE_DELAY = 10
+Entry = namedtuple('EntityRegistryEntry',
+                   'entity_id,unique_id,platform,domain')
```

Comment:
Would it be worth using attrs instead of namedtuples in general, see http://www.attrs.org/en/stable/why.html#namedtuples

Reply:
I don't want to add another dependency. If anything, we would use a backport of the [data classes](https://hackernoon.com/a-brief-tour-of-python-3-7-data-classes-22ee5e046517) that are being added to Python 3.7.

For the current use case namedtuples will be enough.